### PR TITLE
#57 IdentifierTypeCode Binding Change

### LINKED ISSUE #57: Patient identifier type codes are missing in the valueSet
**Real Submitter**
Roland Maurer, pineIT

**Resource(s)/Profiles incl. Version**
HL7 AT FHIR Core IG STU1, Patient Profile

**Links**
[https://fhir.hl7.at/r4-core-STU1/StructureDefinition-at-core-patient.html](https://fhir.hl7.at/r4-core-STU1/StructureDefinition-at-core-patient.html)

**Priority**
Major

**Describe the bug**
The Patient.identifier:XXX.type.coding.code values that are used for the SSNr, bPK and local-PID are part of the codeSystem but not of the valueSet that is recommended by the FHIR standard, see http://hl7.org/fhir/r4/datatypes.html#Identifier.
This causes no problems for resource instances since they reference to the codeSystem but it causes problems for validating the profile. Since the recommended valueSet (http://hl7.org/fhir/r4/valueset-identifier-type.html) is extensible the needed codes from the codeSystem should be added.

**To Reproduce**
see description

**Expected behavior**
The used codes for Patient.identifier:XXX.type.coding.code are part of the valueSet.

**Attachments**
-

**Additional context**
-

### PR DESCRIPTION
**Pull request by regelkraut**
Changed the binding of identifier.type to the new ELGA ValueSet for identifierTypeCodes

Closes #57

### CODE REVIEW

File: input/fsh/at-core-patient.fsh [hunk 1]
```diff
@@ -18,12 +18,15 @@ The HL7® AT Core Patient is based upon the core FHIR® Patient Resource and des
 * address only HL7ATCoreAddress
 * contact.address only HL7ATCoreAddress
 
+// Changing the ValueSet binding for the type of identifiers
+* identifier.type from https://termgit.elga.gv.at/ValueSet/elga-patientidentifier (extensible)
```
Comment (Nik-xT) [inline input/fsh/at-core-patient.fsh:22]: ValueSet was renamed, please change to
```suggestion
* identifier.type from https://termgit.elga.gv.at/ValueSet/hl7-at-patientidentifier (extensible)
```

File: input/fsh/at-core-patient.fsh [hunk 2]
```diff
@@ -18,12 +18,15 @@ The HL7® AT Core Patient is based upon the core FHIR® Patient Resource and des
...
 
+// Changing the ValueSet binding for the type of identifiers
+* identifier.type from https://termgit.elga.gv.at/ValueSet/elga-patientidentifier (extensible)
 //Slice for Identifier, order matters
 * identifier ^slicing.rules = #open
 * identifier ^slicing.discriminator.type = #value
 * identifier ^slicing.discriminator.path = "type.coding.code"
 * identifier ^slicing.ordered = false
 * identifier contains socialSecurityNumber 0..1 and bPK 0..* and localPatientId 0..1
+* identifier[socialSecurityNumber].type from https://termgit.elga.gv.at/ValueSet/elga-patientidentifier (extensible)
```
Comment (Nik-xT) [inline input/fsh/at-core-patient.fsh:29]: ValueSet was renamed, please change to
```suggestion
* identifier[socialSecurityNumber].type from https://termgit.elga.gv.at/ValueSet/hl7-at-patientidentifier (extensible)
```

File: input/fsh/at-core-patient.fsh [hunk 3]
```diff
@@ -32,6 +35,7 @@ The HL7® AT Core Patient is based upon the core FHIR® Patient Resource and des
 * identifier[socialSecurityNumber].system ^short = "OID for the Social Security Number in Austria"
 //* identifier[socialSecurityNumber].assigner.reference = "https://www.gesundheit.gv.at/OID_Frontend/oiddetail.htm?smallView=true&actualOid=1.2.40.0.10.1.4.3.1" (exactly)
 * identifier[socialSecurityNumber].assigner.display = "Dachverband der österreichischen Sozialversicherungsträger" (exactly)
+* identifier[bPK].type from https://termgit.elga.gv.at/ValueSet/elga-patientidentifier (extensible)
```
Comment (Nik-xT) [inline input/fsh/at-core-patient.fsh:38]: ValueSet was renamed, please change to
```suggestion
* identifier[bPK].type from https://termgit.elga.gv.at/ValueSet/hl7-at-patientidentifier (extensible)
```

File: input/fsh/at-core-patient.fsh [hunk 4]
```diff
@@ -40,6 +44,7 @@ The HL7® AT Core Patient is based upon the core FHIR® Patient Resource and des
 * identifier[bPK].system ^short = "OID for the bPK in Austria"
 //* identifier[bPK].assigner.reference = "https://www.gesundheit.gv.at/OID_Frontend/oiddetail.htm?smallView=true&actualOid=1.2.40.0.10.2.1.1.149" (exactly)
 * identifier[bPK].assigner.display = "Bundesministerium für Inneres" (exactly)
+* identifier[localPatientId].type from https://termgit.elga.gv.at/ValueSet/elga-patientidentifier (extensible)
```
Comment (Nik-xT) [inline input/fsh/at-core-patient.fsh:47]: ValueSet was renamed, please change to
```suggestion
* identifier[localPatientId].type from https://termgit.elga.gv.at/ValueSet/hl7-at-patientidentifier (extensible)
```

**Review by Nik-xT — CHANGES_REQUESTED** [review]
Renaming the ValueSet came from the last discussion in TC Terminology. ValueSet is already deleted with its old name and published with its new one.

**Review by Nik-xT — APPROVED** [review]
Looking good!
The first step of getting valuesets & codesystems out of the IG to a TX like termgit.

**Review by Nik-xT — APPROVED** [review]

**Comment by regelkraut** [comment]
@Nik-xT  I had to adapt the binding once more, but only on the level of the actual slices.
For some reason, in order to be applied by Sushi, I had to strengthen the binding from "extensible" to "required" (but just on the slice level). Hope that change is still ok for you as reviewer.

**Comment by aschuler84** [comment]
> @Nik-xT I had to adapt the binding once more, but only on the level of the actual slices. For some reason, in order to be applied by Sushi, I had to strengthen the binding from "extensible" to "required" (but just on the slice level). Hope that change is still ok for you as reviewer.

@regelkraut IMHO this makes sense, otherwise it would be possible to have codes not available in the binded valueset which would make discrimination of the slice type impossible. Seems as if sushi verifies this, not sure though?

**Review by aschuler84 — APPROVED** [review]
Looking good from my side, no further changes.

**Pull request merged**